pkg/odo/cli/pipelines/ui/ui.go

```diff
@@ -236,3 +251,97 @@ func SelectOptionCommitStatusTracker() string {
 	return optionCommitStatusTracker
 }
 
+// check if the length of secret is less than 16 chars
+func CheckSecretLength(secret string) bool {
+	if secret != "" {
+		if len(secret) < 16 {
+			return true
+		}
+	}
+	return false
+}
+
+// ValidatePrefix checks the length of the prefix with the env crosses 63 chars or not
+func ValidatePrefix(prefix string) survey.Validator {
+	return func(input interface{}) error {
+		if s, ok := input.(string); ok {
+			err := validation.ValidateName(s)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		return nil
+	}
+}
+
+// validateSecretLength validates the length of the secret
+func validateSecretLength(secret string) survey.Validator {
+	return func(input interface{}) error {
+		if s, ok := input.(string); ok {
+			err := CheckSecretLength(s)
+			if err == true {
+				return fmt.Errorf("The secret length should 16 or more ")
+			}
+			return nil
+		}
+		return nil
+	}
+}
+
+// validateOverwriteOption(  validates the URL
+func validateOverwriteOption(path string) survey.Validator {
+	return func(input interface{}) error {
+		if s, ok := input.(string); ok {
+			if s == "no" {
+				exists, _ := ioutils.IsExisting(ioutils.NewFilesystem(), filepath.Join(path, "pipelines.yaml"))
+				if exists {
+					EnterOutputPath()
+				}
+			}
+			return nil
+		}
+		return nil
+	}
+}
+
+// validateAccessToken validates if the access token is correct for a particular service repo
+func validateAccessToken(serviceRepo string) survey.Validator {
+	return func(input interface{}) error {
+		if s, ok := input.(string); ok {
+			repo, _ := git.NewRepository(serviceRepo, s)
+			parsedURL, err := url.Parse(serviceRepo)
+			repoName, err := git.GetRepoName(parsedURL)
+			_, _, err = repo.Client.Repositories.Find(context.Background(), repoName)
+			if err != nil {
+				return fmt.Errorf("The token passed is incorrect for repository %s", repoName)
+			}
+			return nil
+		}
+		return nil
+	}
+}
+
+// validateSealedSecretService validates to see if the sealed secret service is present in the correct namespace.
+func validateSealedSecretService(sealedSecretService *types.NamespacedName) survey.Validator {
+	return func(input interface{}) error {
+		if s, ok := input.(string); ok {
+			sealedSecretService.Name = s
+			sealedSecretService.Namespace = EnterSealedSecretNamespace()
+			_, err := secrets.GetClusterPublicKey(*sealedSecretService)
+			if err != nil {
+				if compareError(err, sealedSecretService.Name) {
+					return fmt.Errorf("The given service %s is not installed in the right namespace %s", sealedSecretService.Name, sealedSecretService.Namespace)
```

Comment:
You can use `%q` which will output `The given service "this " is not installed in the right namespace` which looks nicer (and can highlight whitespace issues better).